### ISSUE TITLE
Wrong application of sparse categorical cross entropy loss in tutorial

### ISSUE THREAD

**Issue by Laubeee** [keras-team/keras-io#1147]
Hi, I followed [this tutorial](https://keras.io/examples/vision/oxford_pets_image_segmentation/) and found it trained absolute garbage for my case. After some searching I found the format of the labels is incorrect. The last dimension should be removed, i.e. [here](https://github.com/keras-team/keras-io/blob/master/examples/vision/oxford_pets_image_segmentation.py#L96) and remove [this](https://github.com/keras-team/keras-io/blob/master/examples/vision/oxford_pets_image_segmentation.py#L99).

Of course even nicer would be nice if  SparseCategorical could just handle the additional dimension properly, it should be possible given the output shape of the net?

version 2.10.0

**Comment by sachinprasadhs** [comment]
Hi,

The linked tutorial is to demonstrate the API usage, but not to prove any benchmark results.

Feel free to to fine-tune the network as per your use case. Thanks!

**Comment by Laubeee** [comment]
phew, it's been a while, and my links to the code seem no longer to point to anything useful unfortunately.

If I remember correctly, the tutorial was simply wrong in how it used the data.
IMO it is misleading for a tutorial, especially for the sake of API usage, to use incorrect dimensions resulting in a net that doesn't actually learn something useful, hence I reported this here.

However, I won't pursue this further (i.e. check if the issue still persists) as it's no longer relevant for me.
